.github/workflows/publish.yml

```diff
@@ -0,0 +1,41 @@
+name: Nix Continuous Deployment
+on: [push]
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      deployments: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: "18"
+
+      - name: Install dependencies
+        run: sudo apt -y update && sudo apt -y install libusb-1.0-0-dev libudev-dev
+
+      - name: Setup Foundry
+        uses: foundry-rs/foundry-toolchain@v1
+        with:
+          version: nightly
+
+      - name: Yarn Install
+        run: yarn install --mode=skip-build && yarn allow-scripts
+
+      - name: Setup Forge
+        run: yarn workspace @ubiquity/contracts forge:install
+
+      - name: Build All
+        run: yarn build
+
+      - name: Nix CI Deploy
+        run: |
+          cd packages/dapp/dist
+          npm i -g @railway/cli
```

Comment:
Also global install is unnecessary you should use npx

Reply:
Perhaps something like 

```
npx @railway/cli link
npx @railway/cli up
```

Reply:
Is there any benefit of using npx over npm in the following context since npx also installs the package first? @pavlovcik

Reply:
Functionally speaking, in this context no. Generally speaking, as in developer advice/best practices I think its a bad idea to globally install anything. It leads to too many issues when collaborating on teams with other environments.